indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -101,7 +112,52 @@ public ForkingTaskRunner(
     this.node = node;
     this.portFinder = new PortFinder(config.getStartPort());
 
-    this.exec = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(workerConfig.getCapacity()));
+    this.exec = MoreExecutors.listeningDecorator(
+        Executors.newFixedThreadPool(
+            workerConfig.getCapacity(),
+            new ThreadFactoryBuilder()
+                .setNameFormat("forking-task-runner-%d")
+                .setDaemon(true)
+                .build()
+        )
+    );
+  }
+
+  @Override
+  public List<Pair<Task, ListenableFuture<TaskStatus>>> restore()
+  {
+    final File restoreFile = getRestoreFile();
+    final TaskRestoreInfo taskRestoreInfo;
+    if (restoreFile.exists()) {
+      try {
+        taskRestoreInfo = jsonMapper.readValue(restoreFile, TaskRestoreInfo.class);
+      }
+      catch (Exception e) {
+        log.warn(e, "Failed to restore tasks from file[%s]. Skipping restore.", restoreFile);
```

Comment:
this should be error as restoreFile existence should mean that there are valid tasks to restore and for whatever reason reading the file failed.